[26기 김유신] [Mission 1, 2] 로그인 페이지, 로그인 버튼 활성화 완료

kimnew6 opened the pull request:
- 로그인 페이지 레이아웃 작성 완료
- 로그인 페이지 유효성 검사 기능 관련 로직 작성 완료
      [Mission 2] ID, PW 입력 시 로그인 버튼 활성화 기능 구현
-ID, PW 각1자리 이상 입력시 로그인 버튼 활성화

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인 페이지 레이아웃 작성 완료
- 로그인 버튼 활성화 기능 구현
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id 포함, pw 1 글자 이상 입력 시 버튼 활성화

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 마음처럼 안 되네요

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Jun4928 commented:
라벨 달아주세요. 템플릿이 지켜지지 않으면 리뷰가 진행되지 않습니다!

Jun4928 commented on students/13th/kimsuyeon/login.html:
```diff
@@ -0,0 +1,28 @@
+    <script defer src="js\login.js"></script>
+  </head>
+  <body>
+    <main>
+      <section>
+        <div id="logo">
+          <img src="img\logo_text.png" alt="logo" />
+        </div>
+        <div id="login">
+          <input type="text" placeholder="전화번호, 사용자 이름 또는 이메일" />
+          <input type="password" placeholder="비밀번호" />
+          <button>로그인</button>
+        </div>
+        <div id="alert">
+          <p>비밀번호를 잊으셨나요?</p>
+        </div>
+      </section>
+    </main>
```
semantic tag 사용 GOOD! 👍

Jun4928 commented on students/13th/kimsuyeon/main.html:
```diff
@@ -0,0 +1,208 @@
+    <header>
+      <nav id="nav_bar">
+        <div class="logo">
+          <img
+            class="camera"
+            src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/logo.png"
+          />
+          <img src="img\logo_text.png" />
```
img 태그에는 **alt** 속성을 채워주는 것이 좋습니다! 
이미지가 로딩되지 않았을 때, alt 속성에 넣어준 값이 대신 화면에 렌더링 됩니다.

Link717 replied:
준님, 예를 들어  feed image 처럼 매번 내용을 정확히 알 수 없는 이미지의 경우에는 grouping 할 수 있는 이름으로 표현하는 것도 괜찮을까요?

Jun4928 commented on students/13th/kimsuyeon/main.html:
```diff
@@ -0,0 +1,208 @@
+          <img src="img\logo_text.png" />
+        </div>
+        <input class="serch" type="text" placeholder="검색" />
+        <div class="icons">
+          <ul class="icons">
```
class name 의 중복을 피해주세요. CSS 에서 class 이름으로 해당 html 태그에 접근하는 경우가 많습니다.

Link717 replied:
네, 수정했습니다~

Jun4928 commented on students/13th/kimsuyeon/main.html:
```diff
@@ -0,0 +1,208 @@
+        <div class="feeds">
+          <article>
+            <div class="user_info">
+              <img
+                class="user_image"
+                src="https://scontent-gmp1-1.cdninstagram.com/v/t51.2885-19/s320x320/115712474_636071880347359_8950118108606968740_n.jpg?_nc_ht=scontent-gmp1-1.cdninstagram.com&_nc_ohc=b8d8Z5mKA9QAX_M6Rmy&oh=6837bb45a04d9a665a0634d8d8dd3fa7&oe=5F9376D7"
+              />
```
alt!

Link717 replied:
네! 추가했습니다~

Jun4928 commented on students/13th/kimsuyeon/main.html:
```diff
@@ -0,0 +1,208 @@
+              />
+              <p class="user_id">kakaofriends_official</p>
+            </div>
+            <img
+              class="feed_img"
+              src="https://scontent-gmp1-1.cdninstagram.com/v/t51.2885-15/sh0.08/e35/s750x750/120111026_682306929047142_5571023128378603620_n.jpg?_nc_ht=scontent-gmp1-1.cdninstagram.com&_nc_cat=106&_nc_ohc=Gi6twwXc50IAX9T6Dh3&oh=94298a49cb9185f4538529be8e321394&oe=5F94C719"
+            />
```
alt!

Link717 replied:
네! 추가했습니다~

Jun4928 commented on students/13th/kimsuyeon/main.html:
```diff
@@ -0,0 +1,208 @@
+            </li>
+          </ul>
+        </div>
+        <footer>
+          <p>소개·도움말·홍보 센터·API·채용 정보</p>
+          <br />
+          <p>개인정보처리방침·약관·위치·인기 계정·해시태그·언어</p>
+          <br />
+          <br />
+          <p>© 2020 INSTAGRAM FROM FACEBOOK</p>
+        </footer>
+      </aside>
```
이런 semantic 태그의 사용 👍
계속해서 습관이 될 수 있도록 코드 작성 해 주세요!

HTML 코드에 대한 가독성을 높여주고,
웹사이트가 후에 배포되었을 때, 검색엔진에 의해 검색될 때 상위에 노출 될 수 있도록 도와줍니다.

Link717 replied:
네~ semantic! 유념할게요 ㅎㅎ

Jun4928 commented on students/13th/kimsuyeon/styles/login.css:
```diff
@@ -0,0 +1,79 @@
+  width: 350px;
+  height: 500px;
+  margin: auto;
+  /*vh = viewport height 부모 tag와 상관없이 보이는 화면 기준으로 높이를 지정하겠다.
+  vh를 사용하지 않고 height를 100%라고하면 parent tag 기준으로 100%로 설정된다.*/
```
github 에 코드를 올릴 때에는 이런 불필요한 주석은 삭제 해 주세요!

Link717 replied:
네, 삭제했습니다~

Jun4928 commented on students/13th/kimsuyeon/styles/login.css:
```diff
@@ -0,0 +1,79 @@
+  vh를 사용하지 않고 height를 100%라고하면 parent tag 기준으로 100%로 설정된다.*/
+}
+
+div#logo {
+  margin: 30px auto;
+  width: 100%;
+  height: auto;
+  text-align: center;
+}
+
+div#login {
+  margin-top: 50px;
+  width: 100%;
+  text-align: center;
+}
```
id로 css 의 속성을 부여하는 것은 위험한 일입니다.
id사용을 지양해 주세요. id 로 스타일링 하는 것은 쎈 우선순위를 갖기 때문에, 다른 html 태그에 영향을 미칠 가능성이 높습니다.

예를들어, 지금은 login 페이지라 상관없을지 모르지만, 만약 수많은 정보(=수많은 html 태그)를 담고있는 웹페이지를 작성하게 된다면 어떻게 될까요?

Jun4928 replied:
이 리뷰는 login.css, main.css 두 파일 모두 공통으로 반영해 주세요!

Link717 replied:
네, 공통으로 수정했습니다~

Jun4928 commented on students/13th/kimsuyeon/styles/login.css:
```diff
@@ -0,0 +1,79 @@
+  height: 36px;
+}
+
+/*:focus :양식 입력 칸 등 포커스를 받은 요소를 나타낸다.
+보통 사용자가 요소를 클릭 또는 탭하거나, 키보드 tab키로 선택했을 때 발동한다. */
```
불필요한 주석 삭제 해 주세요! 🔥

Link717 replied:
네! 주석 삭제완료! :-)

Jun4928 commented on students/13th/kimsuyeon/styles/main.css:
```diff
@@ -0,0 +1,259 @@
+  width: auto;
+  border-top: 1px solid #bbbbbbd5;
+  padding: 10px 20px;
+}
+input.reply_input {
```
셀렉터 간에 빈 줄 넣어주세요

Link717 replied:
네~ 수정했습니다.

Jun4928 commented on students/13th/kimsuyeon/styles/main.css:
```diff
@@ -0,0 +1,259 @@
+
+.logo img {
+  height: 29px;
+}
+input.serch {
+  width: 215px;
+  height: 25px;
+  background-color: #fafafa;
+  border: 2px solid#ededed;
+  border-radius: 3px;
+  font-size: 14px;
+  color: #ededed;
+}
```
**빈 줄!** 넣어주세요

Link717 replied:
네! 수정했어요!

Jun4928 reviewed (commented):
리뷰 확인 및 반영 해 주세요!
중복되는 리뷰가 많습니다. 모두 리팩토링 해서 다시 push 해 주세요 🙌

Jun4928 commented:
중간에 hidden conversations 에 숨겨진 리뷰가 있습니다. 열어서 모두 확인 해 주세요 👍
수고하셨습니다 👍

Link717 commented on students/13th/kimsuyeon/js/login.js:
```diff
@@ -0,0 +1,27 @@
+keydown, keypress를 사용할 경우, 예상보다 key를 한번 더 눌러야 배경색이 변경됨*/
+
+userid_location.addEventListener("keyup", function (e) {
+  const userid = document.getElementById("login").children[0].value;
+  const userpw = document.getElementById("login").children[1].value;
+  const loginbtn = document.getElementById("login").children[2];
```
변수 선언 위치에 따라서 event가 발생한 시점의  값이 달라질 수 있다는 관점에서 그 때의 값을 저장한다는 의미로 담았는데, 함수 외부에서 변수로 선언된 위치 정보는 이벤트와 상관없이 변하지 않는 정보니까 해당 변수의 value 형태를 취하는 형태로 코드를 변경 해보겠습니다!

Link717 commented on students/13th/kimsuyeon/js/login.js:
```diff
@@ -0,0 +1,27 @@
+});
+
+userpw_location.addEventListener("keyup", function (event) {
+  const userid = document.getElementById("login").children[0].value;
+  const userpw = document.getElementById("login").children[1].value;
+  const loginbtn = document.getElementById("login").children[2];
```
네~ 알겠습니다!

Link717 commented on students/13th/kimsuyeon/js/login.js:
```diff
@@ -0,0 +1,27 @@
+  const userpw = document.getElementById("login").children[1].value;
+  const loginbtn = document.getElementById("login").children[2];
+
+  if (userid.length > 0 && userpw.length > 0) {
```
네~ 알겠습니다!

Link717 commented on students/13th/kimsuyeon/js/login.js:
```diff
@@ -0,0 +1,27 @@
+  const userpw = document.getElementById("login").children[1].value;
+  const loginbtn = document.getElementById("login").children[2];
+
+  if (userid.length > 0 && userpw.length > 0) {
```
네, 알겠습니다!

Commit:
Add : [Mission 1] 로그인 페이지 완료
- 로그인 페이지 레이아웃 작성 완료
- 로그인 페이지 유효성 검사 기능 관련 로직 작성 완료
      [Mission 2] ID, PW 입력 시 로그인 버튼 활성화 기능 구현
-ID, PW 각1자리 이상 입력시 로그인 버튼 활성화
